compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/optimizer/BIRRecordValueOptimizer.java

```diff
@@ -192,7 +192,8 @@ private boolean containsOnlyConstantLoad(BIRNode.BIRFunction defaultFunction) {
         }
         BIRNode.BIRBasicBlock firstBB = defaultFunction.basicBlocks.get(0);
         BIRNode.BIRBasicBlock secondBB = defaultFunction.basicBlocks.get(1);
-        if (!secondBB.instructions.isEmpty() || secondBB.terminator.kind != InstructionKind.RETURN) {
+        if (!secondBB.instructions.isEmpty() || secondBB.terminator.kind != InstructionKind.RETURN ||
+                firstBB.terminator.kind == InstructionKind.CALL) {
```

Comment:
And also, shall we add a test for this?

Reply:
We have a sample test case in https://github.com/ballerina-platform/ballerina-lang/issues/43531. But I could not find a proper location to add the test. Could you suggest a location to add the test?  I will update the PR with the test as soon as we finalise it.

Reply:
Added tests in 017c4bcfacc0ee63616acd47cfe89e610b32b8ea.